inc/toolbox.class.php

```diff
@@ -2676,22 +2676,28 @@ static function cleanTagOrImage($content, array $tags) {
     * Because json can have been modified from addslashes_deep
     *
     * @param string $encoded Encoded JSON
+    * @param boolean $assoc  assoc parameter of json_encode native function
+    * @param boolean $debug  generate a debug message in log
     *
     * @return mixed
     */
-   static public function jsonDecode($encoded) {
+   static public function jsonDecode($encoded, $assoc = false, $debug = true) {
       if (!is_string($encoded)) {
-         Toolbox::logDebug('Only strings can be json to decode!');
+         if ($debug) {
+            Toolbox::logDebug('Only strings can be json to decode!');
+         }
          return $encoded;
       }
 
-      $json = json_decode($encoded);
+      $json = json_decode($encoded, $assoc);
 
       if (json_last_error() != JSON_ERROR_NONE) {
          //something went wrong... Try to stripslashes before decoding.
-         $json = json_decode(self::stripslashes_deep($encoded));
+         $json = json_decode(self::stripslashes_deep($encoded), $assoc);
          if (json_last_error() != JSON_ERROR_NONE) {
-            Toolbox::logDebug('Unable to decode JSON string! Is this really JSON?');
+            if ($debug) {
```

Comment:
Same here. If we're expecting a json and there is something wrong; there is certainly a bug somewhere